include/fruit/component.h

```diff
@@ -161,7 +161,7 @@ class PartialComponent {
    *
    * class MyImplementation {
    * public:
-   *   INJECT(MyImplementation()) {}
+   *   INJECT(MyImplementation()) = default;
```

Comment:
It's true that this is equivalent, but here the constructor is empty just as a minimal example, and `{}` is the more concise option.
I think the `= default` might make some people wonder if this only works with `= default` constructors for some reason.
Not a big deal, but I would lean on leaving this as it was.

Reply:
The problem I see is that some linters (like `clang-tidy`) usually complain about empty body default constructor or destructor. There are some differences between empty body and defaulted constructor, but they are not relevant to your code, because your code has virtual functions.

If you prefer, I can revert all `= default;` that I added, please just let me know what you prefer.

Reply:
Also, please consider CPP Core Guidelines regarding this issue: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-eqdefault